manager/manager.go

```diff
@@ -100,6 +110,34 @@ func (man *volumeManager) Create(volume *types.VolumeInfo) (*types.VolumeInfo, e
 }
 
 func (man *volumeManager) Delete(name string) error {
+	volume, err := man.Get(name)
+	if err != nil {
+		return err
+	}
+
+	controller := volume.Controller
+	if controller != nil {
+		if controller.Running {
+			if err := man.orc.StopInstance(controller.ID); err != nil {
+				return errors.Wrapf(err, "error stopping controller container %s, volume '%s'", controller.ID, volume.Name)
+			}
+		}
+		if err := man.orc.RemoveInstance(controller.ID); err != nil {
+			return errors.Wrapf(err, "error removing controller container %s, volume '%s'", controller.ID, volume.Name)
+		}
+	}
```

Comment:
You might want to call `man.doDetach(volume)` here instead of this `if controller != nil {...}` clause above